provider/datagen/src/lib.rs

```diff
@@ -671,12 +671,13 @@ fn test_keys() {
 #[cfg(feature = "legacy_api")]
 fn test_keys_from_file() {
     #![allow(deprecated)]
+
+    let mut buffer = std::env::var("CARGO_MANIFEST_DIR")
+        .or(core::env!("CARGO_MANIFEST_DIR"))
+        .expect("CARGO_MANIFEST_DIR must be set to run datagen tests");
+    buffer.push_str("/tests/data/tutorial_buffer+keys.txt");
```

Comment:
Nit: You changed some of the other examples to use `Path::join`; please be consistent and use that here as well. Note that `keys_from_file` takes an `AsRef<Path>`.